scripts/core/ui/components/SelectUser.tsx

```diff
@@ -118,47 +117,23 @@ export class SelectUser extends SuperdeskReactComponent<IPropsSelectUser, IState
                 inlineLabel={true}
                 labelHidden={true}
                 getItems={(searchString) => {
-                    this.abortController?.abort();
-                    this.abortController = new AbortController();
-
-                    let query = {$and: []};
-                    const desk = sdApi.desks.getDeskById(this.props.deskId);
-                    const deskMemberIds = (desk?.members ?? []).map((member) => member.user);
-
-                    if (this.props.deskId != null && this.props.deskId != '') {
-                        query.$and.push({_id: {$in: deskMemberIds}});
-                    }
+                    const deskMembers = sdApi.desks.getDeskMembers(this.props.deskId);
 
                     if (searchString != null && searchString.length > 0) {
-                        query.$and.push(getUserSearchMongoQuery(searchString));
-                    }
+                        const query = getUserSearchMongoQuery(searchString);
```

Comment:
Your filtering here only supports a subset of mongo query, ($or operator). How about next to `getUserSearchMongoQuery` you put `matchUserByMongoQuery(user: IUser): boolean` so if someone changes the query they would also see client side matching code and would update it too.

Reply:
done